### DISCUSSION TITLE
start date changes while using ListDataset

### DISCUSSION THREAD

**Discussion by mohammednaseefop** [awslabs/gluonts#2086] [Q&A] [answered]
I'm trying to create dataset as follows
`alist=ListDataset([{'item_id': '10093200_R1_18817000',
 'start': '2015-06-01 00:00:00',
 'target': [3259.2,
  4480.0,
  5024.25,
  3292.8,
  3500.0,
  3500.0,
  1785.0,
  5398.05,
  4060.0]}],freq='7D')`
 
when I try to print this as follows

`list(data)`

`[{'item_id': '10093200_R1_18817000', 'start': Timestamp('2015-05-28 00:00:00', freq='7D'), 'target': array([3259.2 , 4480. , 5024.25, 3292.8 , 3500. , 3500. , 1785. , 5398.05, 4060. ], dtype=float32), 'source': SourceContext(source='list_data', row=0)}]`

I observed that the start date in the dataset changed from 2015-06-01 to 2015-05-28. Why is it happening?
I really appreciate any help you can provide.

**Comment by jaheba** [comment, marked as answer]
The dates are rolled-back such that each day in a given week is turned into the same date.

Note that we change this behaviour in the upcoming release of `0.10`. There we use `pd.Period` instead of `pd.Timestamp` and don't align dates.

**Reply by mohammednaseefop** [reply]
"The dates are rolled back such that each day in a given week is turned into the same date."
@jaheba, Can you elaborate on this? What exactly is being done here?
When I change the freq, the start date changes to a different value.

**Reply by jaheba** [reply]
If you take monthly frequency as an example, the idea is that any day within that month is turned into the same date value. If you would choose daily frequency instead, each date value would just be aligned for the day, etc.

This works well as long as you have a natural boundary, a month or day in the examples. However, for multiples this starts to get weird, where do you anchor three days? Thus in `0.10` we rely on `pandas.Period` to do the alignment for us. There, it only resets using the base frequency:

```py
>>> p1 = pd.Period("2020-01-01", freq="3d")
>>> p1 + 1
Period('2020-01-04', '3D')

>>> p2 = pd.Period("2020-01-02", freq="3d")
>>> p2 + 1
Period('2020-01-05', '3D')
```

**Reply by mohammednaseefop** [reply]
Thanks! @jaheba for the reply
When can we expect the release of the `0.10`?

**Reply by jaheba** [reply]
It's planned to be released on June 30: https://github.com/awslabs/gluon-ts/milestone/9

You can install it already via:

```sh
pip install gluonts==0.10.0rc1
```
